packages/@o3r/analytics/schematics/ng-add/index.ts

```diff
@@ -1,11 +1,22 @@
 import type { Rule } from '@angular-devkit/schematics';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import type { NgAddSchematicsSchema } from './schema';
 
 /**
  * Add Otter analytics to an Angular Project
  *
  * @param options
  */
-export function ngAdd(): Rule {
+export function ngAdd(options: NgAddSchematicsSchema): Rule {
   /* ng add rules */
-  return () => {};
+  return async (tree) => {
+    const { ngAddPackages, getProjectFromTree } = await import('@o3r/schematics');
+    const packageJsonPath = path.resolve(__dirname, '..', '..', 'package.json');
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf-8' }));
+    const workspaceProject = tree.exists('angular.json') ? getProjectFromTree(tree, options.projectName) : undefined;
```

Comment:
```suggestion
    const workspaceProject = getProjectFromTree(tree, options.projectName);
```

Reply:
I've updated it according to https://github.com/AmadeusITGroup/otter/pull/811